### PR TITLE
[AskMode] Fix `GetOverlap` method of `SortedSpans`

### PR DESCRIPTION
**Pull request by genlu**
The result is correctly preserved color for error message when copied from interactive window.
Fix https://github.com/dotnet/roslyn/issues/6100.

@dotnet/interactive

### CODE REVIEW

File: src/InteractiveWindow/Editor/Output/SortedSpans.cs [hunk 1]
```diff
@@ -50,19 +50,36 @@ public IEnumerable<Span> GetOverlap(Span span)
...
                 if (startIndex < 0)
                 {
-                    return Enumerable.Empty<Span>();
+                    startIndex = ~startIndex;
                 }
+                                                       
+                if (startIndex > 0)
+                {
+                    startIndex = startIndex - 1;
+                }
```
Comment (cston) [inline src/InteractiveWindow/Editor/Output/SortedSpans.cs]: Is this case needed always or only when ```BinarySearch returns a negative number```? In other words, should this move inside the ```if``` above?

File: src/InteractiveWindow/EditorTest/SortedSpansTests.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            // overlap with [0, 25) 
+            Assert.Equal(new Span[] { new Span(10, 10) },
+                         spans.GetOverlap(new Span(0, 25)));
+            // overlap with [25, 45) 
+            Assert.Equal(new Span[] { new Span(30, 10) },
+                         spans.GetOverlap(new Span(25, 20)));
+            // overlap with [0, 50)
+            Assert.Equal(new Span[] { new Span(10, 10), new Span(30, 10) },
+                         spans.GetOverlap(new Span(0, 50)));
+        }
```
Comment (cston) [inline src/InteractiveWindow/EditorTest/SortedSpansTests.cs]: Do we need tests for ```overlap with [*, 0)```?

File: src/InteractiveWindow/Editor/Output/SortedSpans.cs [hunk 3]
```diff
@@ -50,19 +50,36 @@ public IEnumerable<Span> GetOverlap(Span span)
...
+
+                Debug.Assert(startIndex >= 0);
 
                 int spanEnd = span.End;
-                for (int i = startIndex; i < _spans.Count && _spans[i].Start < spanEnd; i++)
+                for (int i = startIndex; i < count && _spans[i].Start < spanEnd; i++)
                 {
                     var overlap = span.Overlap(_spans[i]);
                     if (overlap.HasValue)
```
Comment (cston) [inline src/InteractiveWindow/Editor/Output/SortedSpans.cs]: Can we add ```Debug.Assert(overlap.HasValue)``` as well?

File: src/InteractiveWindow/EditorTest/SortedSpansTests.cs (hunk 2 above)
Comment (cston) [inline src/InteractiveWindow/EditorTest/SortedSpansTests.cs]: Do we need tests for ```overlap with Span(*, 0)```?

Comment (genlu) [inline src/InteractiveWindow/EditorTest/SortedSpansTests.cs]: what is *?

File: src/InteractiveWindow/Editor/Output/SortedSpans.cs (hunk 1 above)
Comment (genlu) [inline src/InteractiveWindow/Editor/Output/SortedSpans.cs]: Yes, this is only needed when a negative number returned. I will move it.

**Comment by MattGertz** [comment]
Approved pending successful test runs.

File: src/InteractiveWindow/EditorTest/SortedSpansTests.cs (hunk 2 above)
Comment (genlu) [inline src/InteractiveWindow/EditorTest/SortedSpansTests.cs]: Added.

**Comment by genlu** [comment]
Test win, please

**Comment by cston** [comment]
LGTM

**Comment by genlu** [comment]
test this please.

**Comment by genlu** [comment]
Test Linux please.

**Pull request merged**